pkg/ir/derivation.go

```diff
@@ -161,6 +161,11 @@ func fillCSVFieldDesc(cellData string, fieldDescMap FieldDescMap, fieldName stri
 		// implicit set shape if shape is not provided in SQL COLUMN clause
 		fieldDescMap[fieldName].Shape = []int{len(values)}
 	}
+
+	if fieldDescMap[fieldName].IsSparse {
+		fieldDescMap[fieldName].DType = Int // force to set sparse tensor in CSV format to be of type Int
```

Comment:
Why? We currently do not have test cases that use sparse float tensors, should add FIXME comment here maybe.

Reply:
In fact, it is the previous logic.

https://github.com/sql-machine-learning/sqlflow/blob/73cda1925d13cc92b08a7864018708ff988f5685/pkg/ir/ir_generator.go#L889-L892

I guess it is because we only support sparse tensor encoded in CSV format currently, and the numeric values in the cell data in CSV format are the indices where the tensor values are 1. For example, "3,5,7" indicates a sparse tensor `x`, and `x[3] = x[5] = x[7] = 1`, and the other values of `x` are 0. Since the index is always integral, the sparse tensor encoded in CSV format would be tensor of integers. 

I add a `FIXME` comment, and we should remove this constraint if we will support other sparse tensor data format in the future.

Reply:
Cool. Thanks